synapse/lib/filepath.py

```diff
@@ -0,0 +1,530 @@
+    lists of paths are joined prior to parsing
+    The path supports python's fnmatch glob matching
+
+    '''
+    if None in paths:
+        return None
+
+    path = genpath(*paths)
+
+    path_parts = getPathParts(path)
+
+    cls = _pathClass(path_parts[0])
+    base = cls(path_parts[0], parent=None)
+
+    bases = [base]
+    chld_bases = []
+    for name in path_parts[1:]:
+        chld_bases = []
+        for base in bases:
+            for member in base.list():
+                if not fnmatch.fnmatch(member, name):
+                    continue
+                chld_bases.append(base.child(member))
+
+        bases = chld_bases
+
+    if not bases:
+        return None
+
+    return bases
+
+def parsePath(*paths):
+    '''
+    function to parse the incoming path.
+    lists of paths are joined prior to parsing
+    '''
+    if None in paths:
+        return None
+
+    path = genpath(*paths)
+
+    path_parts = getPathParts(path)
+
+    try:
+
+        cls = _pathClass(path_parts[0])
+        base = cls(path_parts[0], parent=None)
+
+        for name in path_parts[1:]:
+            base = base.child(name)
+            if base == None:
+                return None
+
+    except s_exc.NoSuchPath as e:
+        return None
+
+    return base
+
+def openFiles(*paths, **kwargs):
+    '''
+    Yields a read-only file-like object for each path even if the path terminates inside a container file.
+    Paths may use python's fnmatch glob matching
+
+    If the path is a regular os accessible path mode may be passed through as a keyword argument.
+    If the path terminates in a container file, mode is ignored.
+
+    If the path does not exist a NoSuchPath exception is raised.
+    If req=True (Default) NoSuchPath will also be raised if ANY matching path exists, but is a directory
+
+    ex.
+    openFiles('/foo/bar/*.egg/dir0/zz*/nest.zip')
+    '''
+    reqd = kwargs.get('req', True)
+    mode = kwargs.get('mode', 'r')
+    fpaths = parsePaths(*paths)
+
+    if not fpaths:
+        raise s_exc.NoSuchPath(path='%r' % (paths,))
+    for path in fpaths:
+        if not path.isfile():
+            if not reqd:
+                continue
+            raise s_exc.NoSuchPath(path=path.path())
+        yield path.open(mode=mode)
+
+def openFile(*paths, **kwargs):
+    '''
+    Returns a read-only file-like object even if the path terminates inside a container file.
+
+    If the path is a regular os accessible path mode may be passed through as a keyword argument.
+    If the path terminates in a container file, mode is ignored.
+
+    If the path does not exist a NoSuchPath exception is raised.
+    If req=True (Default) NoSuchPath will also be raised if the path exists, but is a directory
+
+    ex.
+    openFile('/foo/bar/baz.egg/path/inside/zip/to/file')
+    '''
+    reqd = kwargs.get('req', True)
+    mode = kwargs.get('mode', 'r')
+    fpath = parsePath(*paths)
+
+    if not fpath:
+        raise s_exc.NoSuchPath(path='%r' % (paths,))
```

Comment:
Any convention for the value of a kwarg on a SynErr exception should probably maintain consistency with the type/format for that data as well.  In this case, we should probably say path='/'.join(paths) or something so it's a string of the actual path rather than a repr of the tuple